index.js

```diff
@@ -118,7 +166,16 @@ function filterByGenre() {}
       }
     ];
  */
-function getAllMoviesReleasedAtOrBeforeYear() {}
+function getAllMoviesReleasedAtOrBeforeYear(movies, year) {
+  if(movies.length === 0){
+    throw "there are no movies"
+  }
+  let moviesArr = movies.filter((movie) =>{
+    let releasedDates = movie.released;
+    let movieYear = releasedDates.pop();
+    console.log(movieYear)
```

Comment:
You were on the right track with this one!
```suggestion
  let moviesArr = movies.filter((movie) =>{
    let releasedDates = movie.released.split(' '); //split the release date string into an array
    let movieYear = releasedDates.pop(); // .pop is an array method, so it doesn't work on strings.
    return movieYear === year   //if movieYear is equal to the inputted year, "movieYear === year" will be truthy. Only movies that meet this condition will make it into the filtered array.
```